micro-rdk-nmea/src/parse_helpers/parsers.rs

```diff
@@ -0,0 +1,356 @@
+/// as the first n bits of `size_of::<T>()` bytes with the remaining bits as zeroes (the resulting bytes
+/// will be parsed as Little-Endian). See invocation of the number_field macro below for the currently
+/// supported number types.
+pub struct NumberField<T> {
+    bit_size: usize,
+    _marker: PhantomData<T>,
+}
+
+impl<T> NumberField<T> {
+    pub fn new(bit_size: usize) -> Result<Self, NumberFieldError> {
+        let max_bit_size = std::mem::size_of::<T>() * 8;
+        if bit_size > max_bit_size {
+            Err(NumberFieldError::ImproperBitSize(bit_size, max_bit_size))
+        } else {
+            Ok(Self {
+                bit_size,
+                _marker: Default::default(),
+            })
+        }
+    }
+}
+
+impl<T> Default for NumberField<T>
+where
+    T: Sized,
+{
+    fn default() -> Self {
+        Self {
+            bit_size: size_of::<T>() * 8,
+            _marker: Default::default(),
+        }
+    }
+}
+
+macro_rules! number_field {
+    ($($t:ty),*) => {
+        $(
+            impl FieldReader for NumberField<$t> {
+                type FieldType = $t;
+
+                fn read_from_data(&self, data: &[u8], start_idx: usize) -> Result<(usize, Self::FieldType), NmeaParseError> {
+                    let type_size = std::mem::size_of::<$t>();
+                    match self.bit_size {
+                        x if x == (type_size * 8) => {
+                            let value: &[u8] = &data[start_idx..(start_idx + type_size)];
+                            Ok((start_idx + type_size, <$t>::from_le_bytes(value.try_into()?)))
+                        }
+                        x => {
+                            let shift = (type_size * 8) - x;
+                            let end_idx = start_idx + (x / 8);
+                            let value: &[u8] = &data[start_idx..(end_idx + 1)];
+                            let value = <$t>::from_le_bytes(value.try_into()?);
+                            Ok((end_idx, value >> shift << shift))
+                        }
+                    }
+                }
+            }
+        )*
+    };
+}
+
+number_field!(u8, i8, u16, i16, u32, i32, u64, i64);
+
+/// A field reader for parsing data into a field type that implements the `Lookup` trait.
+/// The `bit_size` property is used to parse a raw number value first with a methodology similar to the one
+/// defined in the documentation for NumberField, after which the raw value is passed to `Lookup::from_value`
+pub struct LookupField<T> {
+    bit_size: usize,
+    _marker: PhantomData<T>,
+}
+
+impl<T> LookupField<T> {
+    pub fn new(bit_size: usize) -> Self {
+        Self {
+            bit_size,
+            _marker: Default::default(),
+        }
+    }
+}
+
+impl<T> FieldReader for LookupField<T>
+where
+    T: Lookup,
+{
+    type FieldType = T;
+
+    fn read_from_data(
+        &self,
+        data: &[u8],
+        start_idx: usize,
+    ) -> Result<(usize, Self::FieldType), NmeaParseError> {
```

Comment:
Same thoughts as above on signature here regarding `self` and the `usize` return.